lib/using_restricted_array.rb

```diff
@@ -6,56 +6,124 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: The time complexity is O(n), where n is the size of the array.
+# Space complexity: The space complexity is O(1).
 def length(array)
-  raise NotImplementedError
+  size = 0
+
+  while array[size] != nil
+    size += 1
+  end
+
+  return size
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: The time complexity is O(n), where n is the size of the array.
+# Space complexity: The space complexity is O(n), because the printed_array variable grows as the input grows.
 def print_array(array)
```

Comment:
👍 , although your space complexity is O(1) because you don't save any more data, just print things to the screen.

Reply:
Hey Chris! Just wanted to clarify because the space complexity for this method was confusing me...

So even if I shovel all array elements into the variable printed_array, space complexity is still O(1)?